src/filter_parameters.rs

```diff
@@ -19,8 +25,85 @@ pub struct FilterParameters {
     pub drive: ParameterF32,
     pub mode: ParameterUsize,
     pub slope: ParameterUsize,
-    pub filter_type: AtomicUsize,
+    pub filter_type: ParameterUsize,
+
+    pub reserved: ParameterUsize,
+}
+
+#[repr(i32)]
+#[derive(FromPrimitive, Eq, PartialEq, Debug, EnumIter)]
+pub enum FilterParameterNr {
+    Cutoff,
+    Res,
+    Drive,
+
+    FilterType = 32,
+    Mode,
+    Slope,
+
+    #[num_enum(default)]
+    Padding = 200,
+}
+
+impl GetParameterByIndex for FilterParameters {
+    fn get_parameter_by_index<'a>(&'a self, index: i32) -> &'a dyn Parameter {
+        // if index > 0x10 {
+        //     println!("-- get_parameter_by_index {}", index);
+        // }
+        match FilterParameterNr::from(index) {
+            Cutoff => &self.cutoff,
+            Res => &self.res,
+            Drive => &self.drive,
+            FilterType => &self.filter_type,
+            Mode => &self.mode,
+            Slope => &self.slope,
+            Padding => &self.reserved,
+        }
+    }
+}
+
```

Comment:
Great step in the right direction! The fact that his works makes me think it should be possible to implement the `Index<usize>` trait (the trait used to index e.g. a vector) directly:

```rust
impl Index<usize> for FilterParameters{
    type Output = dyn Parameter;
    fn index(&self, i: usize) -> &Self::Output {
        match FilterParameterNr::from(index) {
            ...
        }
    }
}
```
This would allow doing `self[param_index as usize].get_parameter()`